api/krusty/openapicustomschema_test.go

```diff
@@ -0,0 +1,403 @@
+package krusty_test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+)
+
+func writeTestSchema(th kusttest_test.Harness, filepath string) {
+	bytes, _ := ioutil.ReadFile("testdata/customschema.json")
+	th.WriteF(filepath+"mycrd_schema.json", string(bytes))
+}
+
+func writeTestComponentWithCustomSchema(th kusttest_test.Harness) {
+	writeTestSchema(th, "/app/comp/")
+	openapi.ResetOpenAPI()
+	th.WriteC("/app/comp", `
```

Comment:
I'm trying to get people to use "." as the top level to make the file structure a little simpler and signal that paths don't have to be absolute or use the word `app`.

e.g.
```
th.WriteF("comp/stub.yaml", ...
...
th.WriteK(".", 
...
m := th.Run(".", th.MakeDefaultOptions())
```

Does that seem simpler to you ?

Reply:
done